### ISSUE TITLE
Snippet trigger max length

### ISSUE THREAD

**Issue by xYuusha** [occivink/kakoune-snippets#20]
I have been trying this plugin for a while and it's great but I'm having a problem with trigger lengths. Some commits before, I could use some slightly longer triggers but now they are failing. After checking the source, the reason is that you are calling "h6h", limiting the trigger length to 6.

Is there a good reason why it's set to that number? Is it expected for the triggers to be always short?

**Comment by occivink** [comment]
Allow me to point you to the [FAQ](https://github.com/occivink/kakoune-snippets/blob/master/README.md#whats-with-escaping-what-kind-of-characters-can-i-use-and-not-use). 
I considered that triggers would generally be short so that they can be typed easily. What kind of triggers do you want to use?

**Comment by xYuusha** [comment]
I completely missed out that section on the FAQ, sorry.
Personally, I like triggers a bit more verbose than small chain of letters meaning something.
A simple example should be `function` instead of `fun` or other similar stuff. I do use longer triggers but they are way too personal and I don't think they are a good example.

What I did for my use case is change the `h6hs<ret>` for `h<a-a><a-w>s<ret>`. That allows to check the full word instead of a pre defined length.
It works for me since I don't use spaces but I'm not really experienced with kakoune and I'm not sure if it might break something else.

**Comment by andreyorst** [comment]
> It works for me since I don't use spaces but I'm not really experienced with kakoune and I'm not sure if it might break something else.

I'm not sure if triggers should (or could) contain spaces

**Comment by occivink** [comment]
>I'm not sure if triggers should (or could) contain spaces

I used to think they should be restricted to alphanumeric, but then the topic of a `/*` trigger came up and I've revised my stance to accept any trigger. Spaces should work, currently. Newlines too, if you're feeling particularly adventurous.

As you've observed, `<a-a><a-w>` won't work if triggers contain spaces. The `6h` is kind of a compromise to work with any kind of triggers, with the annoying arbitrary max length.
Ideally we would do an arbitrary reverse search anchored to the current selection, but it's not possible to do efficiently in kakoune currently.

I'll raise it to 20 for now, surely that will be enough.
